src/stcal/jump/jump.py

```diff
@@ -340,50 +379,37 @@ def flag_large_events(gdq, jump_flag, sat_flag, min_sat_area=1,
     :return: none
     """
 
-    log.info('Flagging large events (snowballs, showers).')
+    log.info('Flagging large Snowballs')
 
     n_showers_grp = []
     n_showers_grp_ellipse = []
     for integration in range(gdq.shape[0]):
         for group in range(1, gdq.shape[1]):
-            if use_ellipses:
-                jump_ellipses = find_ellipses(gdq[integration, group, :, :], jump_flag, min_jump_area)
-                n_showers_grp_ellipse.append(len(jump_ellipses))
-                gdq[integration, group, :, :], num_events = \
-                    extend_ellipses(gdq[integration, group, :, :],
-                                    jump_ellipses, sat_flag, jump_flag,
-                                    expansion=expand_factor)
-            else:
-                #  this line needs to be changed later to just look at saturation flags.
-                new_flagged_pixels = gdq[integration, group, :, :] - gdq[integration, group - 1, :, :]
-
-                sat_circles = find_circles(new_flagged_pixels, sat_flag, min_sat_area)
-                sat_pixels = np.bitwise_and(new_flagged_pixels, sat_flag)
-                saty, satx = np.where(sat_pixels == sat_flag)
-                only_jump = gdq[integration, group, :, :].copy()
-                only_jump[saty, satx] = jump_flag
-                jump_circles = find_circles(only_jump, jump_flag, min_jump_area)
-
-                if sat_required_snowball:
-                    snowballs = make_snowballs(jump_circles, sat_circles)
-                else:
-                    snowballs = jump_circles
-                n_showers_grp.append(len(snowballs))
-                gdq[integration, group, :, :], num_events = extend_snowballs(gdq[integration, group, :, :],
-                                                                             snowballs, sat_flag,
-                                                                             jump_flag,
-                                                                             expansion=expand_factor)
-        if use_ellipses:
-            if np.all(np.array(n_showers_grp_ellipse) == 0):
-                log.info(f'No showers found in integration {integration}.')
+            current_gdq = 1.0 * gdq[integration, group, :, :]
+            prev_gdq = 1.0 * gdq[integration, group - 1, :, :]
+            diff_gdq = 1.0 * current_gdq - prev_gdq
+            diff_gdq[diff_gdq != sat_flag] = 0
```

Comment:
i think this line would cause it to miss pixels that were just flagged with a new saturation flag AND some other dq flag. I think lines 388-392 could be replaced with a bitwise & with the saturation flag then subtraction to identify new saturated pixels.

Reply:
Since the groupdq can only get saturated and jump added and only one of them at a time, the subtraction works.